entity-framework/core/what-is-new/ef-core-9.0/whatsnew.md

```diff
@@ -198,153 +139,176 @@ info: 6/10/2024 19:06:00.017 CosmosEventId.ExecutingSqlQuery[30100] (Microsoft.E
+The new `ToPageAsync` operator returns a `CosmosPage`, which exposes a continuation token that can be used to efficiently resume the query at a later point, fetching the next 10 items:
 
-```csharp
-protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
-    => optionsBuilder.ConfigureWarnings(b => b.Ignore(CosmosEventId.SyncNotSupported));
+```c#
+var nextPage = await context.Sessions.OrderBy(s => s.Id).ToPageAsync(10, continuationToken);
 ```
 
-Note, however, that we plan to fully remove sync support in EF11, so start updating to use async methods like `ToListAsync` and `SaveChangesAsync` as soon as possible!
+For more information, [see the documentation section on pagination](xref:core/providers/cosmos/querying#pagination).
 
-### Enhanced primitive collections
+### FromSql for safer SQL querying
 
-> [!TIP]
-> The code shown here comes from [CosmosPrimitiveTypesSample.cs](https://github.com/dotnet/EntityFramework.Docs/tree/main/samples/core/Miscellaneous/NewInEFCore9.Cosmos/CosmosPrimitiveTypesSample.cs).
+The Cosmos DB provider has allowed SQL querying via <xref:Microsoft.EntityFrameworkCore.CosmosQueryableExtensions.FromSqlRaw%2A>. However, that API can be susceptible to SQL injection attacks when user-provided data is interpolated or concatenated into the SQL. In EF 9.0, you can now use the new `FromSql` method, which always integrates parameterized data as a parameter outside the SQL:
 
-The Cosmos DB provider has supported primitive collections in a limited form since EF Core 6. This is support is being enhanced in EF9, starting with consolidation of the metadata and API surfaces for primitive collections in document databases to align with primitive collections in relational databases. This means that primitive collections can now be explicitly mapped using the model building API, allowing for facets of the element type to be configured. For example, to map a list of required (i.e. non-null) strings:
+```c#
+var maxAngle = 8;
+_ = await context.Blogs
+    .FromSql($"SELECT VALUE c FROM root c WHERE c.Angle1 <= {maxAngle}")
+    .ToListAsync();
+```
 
-<!--
-            modelBuilder.Entity<Book>()
-                .PrimitiveCollection(e => e.Quotes)
-                .ElementType(b => b.IsRequired());
--->
-[!code-csharp[ConfigureCollection](../../../../samples/core/Miscellaneous/NewInEFCore9.Cosmos/CosmosPrimitiveTypesSample.cs?name=ConfigureCollection)]
+For more information, [see the documentation section on pagination](xref:core/providers/cosmos/querying#pagination).
+
+### Role-based access
+
+Azure Cosmos DB for NoSQL includes a [built-in role-based access control (RBAC) system](/azure/cosmos-db/role-based-access-control). This is now supported by EF9 for both management and use of containers. No changes are required to application code. See [Issue #32197](https://github.com/dotnet/efcore/issues/32197) for more information.
```

Comment:
Management operations with RBAC aren't supported, see other comments

Reply:
I'm going to leave this out of this PR - I didn't touch this section and am not sure I'm totally familiar with the whole thing... Do we have an issue do make RBAC-related doc changes?